### PR TITLE
MAX_POST_CHARS Env

### PR DESCRIPTION
**Pull request by shleeable**
I completed a quick survey of 10,000+ Mastodon instances and found that 21% of them change their MAX_POST length in some way... that's more than 2100 instance admins messing with the default.... and causing problems/extra work for themselves during upgrades with pointless merge conflicts (for non-glitch admins).

Results here: https://gist.github.com/shleeable/101b131653c0ff0e21a81bf99a31c7f0

As such, I've stolen the code from Glitch for MAX_TOOT_CHARS.... @ClearlyClaire please make sure I'm not missing anything

`RIP TOOT`

### CODE REVIEW

**Comment by shleeable** [comment]
Example of the survey
```
   10335 | TOTAL LINES
     140 | DISTINCT LINES

COUNT    | LINE
-----------------------------------------------------
    8138 | 500
     942 | 5000
     445 | 3000
     135 | 1000
      81 | 10000
      77 | 2000
      49 | 1500
      35 | 100000
```

File: app/serializers/initial_state_serializer.rb
Comment (ClearlyClaire) [inline app/serializers/initial_state_serializer.rb:1]: `max_post_chars` is missing from the `attributes`

File: app/serializers/rest/v1/instance_serializer.rb
Comment (ClearlyClaire) [inline app/serializers/rest/v1/instance_serializer.rb:1]: I wouldn't change this file, as we have the V2 serializer already exporting that bit of information.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]

File: app/serializers/rest/v1/instance_serializer.rb
Comment (shleeable) [reply app/serializers/rest/v1/instance_serializer.rb:1]: change reverted

File: app/serializers/initial_state_serializer.rb
Comment (shleeable) [reply app/serializers/initial_state_serializer.rb:1]: Added attributes

**Comment by shleeable** [comment]
@ClearlyClaire updated.

**Review by ClearlyClaire — APPROVED** [review]

**Comment by vmstan** [comment]
Fixes #12265

**Comment by vmstan** [comment]
I wonder why the rebase required hasn't triggered for this? The initializer file was redone in #28363 and the patch won't apply anymore.

**Comment by shleeable** [comment]
Cite: https://aus.social/@rolle@mementomori.social/111595035041587931

**Comment by adulau** [comment]
Can we expect a merge for the release 4.2.4 ? This would be awesome.

**Comment by ClearlyClaire** [comment]
I don't think @Gargron's stance against making the character limit customizable has changed.

**Comment by jgillich** [comment]
OP does make a strong case for this change in my opinion: 21% of admins already jump through the hoops to change this. It is a frequent request by users, and most other fediverse servers support custom character limits. Gargron is against this because the UI/UX is designed for 500 characters, but you could also mention this in the documentation for this option and make it clear that a limit of 500 is the official recommendation. The majority of admins will stick with the default, but you'll remove pain points for those who'll change it anyhow.

**Comment by PinballsWizard** [comment]
I would be interested to know if the survey data collected also includes the version each instance is running, as compared to the then-current most up to date version.  The additional code changes add complexity to upgrades and simplifying this may be beneficial to ensuring more instance admins update in a more timely manner.

**Comment by shleeable** [comment]
The battle of MAX_CHAR Hill continues.

**Comment by nightpool** [comment]
my stance has always been that I think it's much more healthy for the ecosystem for thousands of admins to learn how to make changes and customize the software they run rather than having to shoulder the complexity of infinite possible customizations in thr base code. additionally I mostly agree wrt the UI. By not adding this to the base mastodon codebase, we encourage the development of alternative server software and client UI that's more purpose built for longer posts, like Tumblr.

**Comment by shleeable** [comment]
> we encourage the development of alternative server software and client UI that's more purpose built for longer posts, like Tumblr.

Me too pal.

**Comment by mwt** [comment]
> > we encourage the development of alternative server software and client UI that's more purpose built for longer posts, like Tumblr.
> 
> Me too pal.
> 
> It's been a dark time for people like me that believe Tumblr is the superior social network.... maybe 2024 is the year of the FediTumblr clone.

or just fedi support in tumblr, which is likely!

**Comment by gabboman** [comment]
> > we encourage the development of alternative server software and client UI that's more purpose built for longer posts, like Tumblr.
> 
> 
> 
> Me too pal. 
> 
> 
> 
> It's been a dark time for people like me that believe Tumblr is the superior social network.... maybe 2024 is the year of the FediTumblr clone.

wafrn added fedederation in 2023 but maybe 2025 will be the year of wafrn (I am trying to be real)

**Comment by cmdr-nova** [comment]
I support a max post toggle, this would be amazing

**Comment by adulau** [comment]
Can we expect a merge for the release 4.2.5 ? This would be awesome. I must admit it will be soon my trigger to move to a fork if it's not finally in Mastodon....

**Comment by arazilsongweaver** [comment]
Most of the UI/UX functionality to support this feature was recently merged in via pull request #28928 ... AFAIK the only thing left to do is to implement the server side of things and clean up any resulting QA/QC items.

**Comment by rahilarious** [comment]
Instead of being dictated by _one_ guy and his decision, Mastodon should use it's own feature and take a **poll** to make decision. 

I shouldn't be needing to compile from source or use fork just for one simple env variable

**Pull request closed without merging**